GeeksCoreLibrary/Modules/Databases/Services/CachedDatabaseHelpersService.cs

```diff
@@ -138,12 +145,43 @@ public async Task DuplicateTableAsync(string tableToDuplicate, string newTableNa
         await databaseHelpersService.DuplicateTableAsync(tableToDuplicate, newTableName, includeData, sourceDatabaseName, destinationTableName);
     }
 
+    /// <inheritdoc />
+    public async Task<Dictionary<string, List<IndexSettingsModel>>> GetIndexesAsync(List<string> tableNames, string databaseName = null)
+    {
+        var cacheName = $"CachedDatabaseHelpersService_GetIndexesAsync_{String.Join("_", tableNames.Order())}_{databaseName}_{branchesService.GetDatabaseNameFromCookie()}";
+        return await cache.GetOrAddAsync(cacheName,
+            async cacheEntry =>
+            {
+                cacheEntry.AbsoluteExpirationRelativeToNow = gclSettings.DefaultQueryCacheDuration;
+                return await databaseHelpersService.GetIndexesAsync(tableNames, databaseName);
+            }, cacheService.CreateMemoryCacheEntryOptions(CacheAreas.Database));
+    }
+
+    /// <inheritdoc />
+    public async Task<List<IndexSettingsModel>> GetIndexesAsync(string tableName, string databaseName = null)
+    {
+        return await GetIndexesAsync(this, tableName, databaseName);
+    }
+
+    /// <inheritdoc />
+    public async Task<List<IndexSettingsModel>> GetIndexesAsync(IDatabaseHelpersService service, string tableName, string databaseName = null)
+    {
+        // Caching is not needed here, because we are already caching the indexes for all tables in the GetIndexesAsync(List<string> tableNames) method.
+        return await databaseHelpersService.GetIndexesAsync(service, tableName, databaseName);
+    }
+
     /// <inheritdoc />
     public async Task CreateOrUpdateIndexesAsync(List<IndexSettingsModel> indexes, string databaseName = null)
     {
         await databaseHelpersService.CreateOrUpdateIndexesAsync(indexes, databaseName);
     }
 
+    /// <inheritdoc />
+    public async Task CreateOrUpdateIndexesAsync(IDatabaseHelpersService service, List<IndexSettingsModel> indexes, string databaseName = null)
+    {
+        await databaseHelpersService.CreateOrUpdateIndexesAsync(service, indexes, databaseName);
```

Comment:
I think the wrong service is used here. It's using the one from the class instead of the passed service.

Reply:
Same as before, this is on purpose.